### PR TITLE
Add validation phone number for subscription

### LINKED ISSUE #5: Set up shared MongoDB account
Since we will eventually be persisting phone numbers to MongoDB, it's my responsibility to set up a shared MongoDB Atlas account so we can all collaborate on this

### PR DESCRIPTION
**Pull request by u5surf**
# ⚠️ __IMPORTANT: Please do not create a Pull Request without creating an issue first.__

All changes need to be discusssed before proceeding. Failure to do so may result in the pull request being rejected.

Please be sure to review our [Code of Conduct](https://github.com/COVID-19-electronic-health-system/.github/blob/master/CODE_OF_CONDUCT.md), [Contributing Guidelines](https://github.com/COVID-19-electronic-health-system/.github/blob/master/CONTRIBUTING.md), and [Support documentation](https://github.com/COVID-19-electronic-health-system/.github/blob/master/SUPPORT.md) before submitting a pull request.

-----

## Please include the issue number the pull request fixes by replacing YOUR-ISSUE-HERE in the text below.

Fixes #5 

## Summary

I added validation phone number when we subscription the service.

## Test Plan (required)
build lamdas/subscrition/main.go and  lamdas/subscrition/main.go
and then, to indicate validation, we'll be executing these to subscribe valid or invalid number case.

## Final Checklist

- [-] For CoronaTracker, did you bump the version in `package.json`?
    - Update the second decimal for a major change
    - Update the third decimal for a minor change
    - Numbers can go past 9, e.g. `1.0.9` => `1.0.10`
    - For more info, read about [Semantic Versioning](https://semver.org/)
- [-] Did you add any new tests as necessary?
- [x] Is your PR rebased off the most current master?
- [x] Have you squashed all commits? _(can be done at merge)_
- [-] Did you use `yarn` not `npm`? _(important!)_
- [-] Did you use Material-UI wherever possible?
- [-] Did you run `yarn lint` on the code?
- [-] Did you run all of your most recent changes locally to make sure everything is working?

### CODE REVIEW

File: lambdas/subscribe/main.go [hunk 1]
```diff
@@ -84,6 +85,14 @@ func Handler(ctx context.Context, request events.APIGatewayProxyRequest) (events
 		}, err
 	}
 
+	matched, err := regexp.MatchString(`^\+[1-9]\d{1,14}$`, requestBody.Number)
+	if err != nil || matched {
```
Comment (acthelemann) [inline lambdas/subscribe/main.go]: Let's create a separate `if` block for `err != nil` that returns a `500` status code similar to how errors are handled throughout this function.

**Review by acthelemann — CHANGES_REQUESTED** [review]
I'm thinking we don't do this check in the unsubscribe lambda, but curious to see what @SomeMoosery thinks

File: lambdas/subscribe/main.go (hunk 1 above)
Comment (u5surf) [reply lambdas/subscribe/main.go]: @acthelemann 
OK, Separated.

**Comment by SomeMoosery** [comment]
So right now we're doing this check in the frontend as far as I know - do we think it could be worthwhile to add in the backend as well as a failsafe? Or does that seem redundant?

**Comment by acthelemann** [comment]
I like the idea of it in both places, but now I'm thinking it might be inconvenient if we want to change the regex/validation since we'll have to do it in both places.

**Comment by SomeMoosery** [comment]
So do you think we should close this for now and reopen if the need arises?

**Pull request closed without merging**